common/defs.go

```diff
@@ -21,10 +21,19 @@ import (
 	"fmt"
 	"net"
 	"strings"
+	"encoding/json"
 )
 
 // Here we only keep type definitions and struct definitions with no behavior.
 
+func String(i interface{}) string {
```

Comment:
It's used by other types that provide a `String() string` method (to satisfy the `fmt.Stringer` interface) -- it's a convenience function to avoid writing repetitive code in those methods.

I think it's a bit odd; produce JSON if possible otherwise emit the default conversion to string.
That doesn't seem helpful for logging, troubleshooting, etc that it would affect.